docs/DeveloperGuide.md

```diff
@@ -2,37 +2,243 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+* This application is built in partial fulfillment of the requirements of CS2113.
+* Knowledge and ideas are adapted from the CS2113 textbook, as well as SE-EDU textbook.
+
+---
+
+## Setting up, getting started
+
+Refer to the User guide at https://ay2324s2-cs2113-w14-2.github.io/tp/UserGuide.html.
+
+---
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Architecture
+
+<img src="images/ArchitectureDiagram.png" width="350" />
+
+The ***Architecture Diagram*** given above explains the high-level design of the application.
+
+Given below is a quick overview of main components and how they interact with each other.
+
+**Main components of the architecture**
+
+**`RecipeIO`**  is in charge of the app launch and shut down.
+* At app launch, it initializes the other components in the correct sequence, and connects them up with each other.
+* At shut down, it shuts down the other components and invokes cleanup methods where necessary.
+
+The bulk of the app's work is done by the following five components:
+
+* `UI` The UI of the App.
+* `RecipeList`: The command executor and holder of data of the app.
+* `InputParser`: Extracts information from the user input into the command line.
+* `CommandValidator`: The command validator.
+* `Storage`: Reads data from, and writes data to, the hard disk.
+
+the `commands`package represents a collection of commands used by **`RecipeList`**
+
+**How the architecture components interact with each other**
+
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+
+<img src="images/DeleteRecipe.png" width="800" />
+
+### RecipeList Component
+
+The **API** of this component is specified in [RecipeList.java](https://github.com/AY2324S2-CS2113-W14-2/tp/tree/master/src/recipeio/recipe/RecipeList.java)
+
+![Structure of the UI Component](images/RecipeListDiagram.png)
+
+The `RecipeList` contains many `Recipe`s, which has attributes:
+* `name` : String
+* `cookTime` : int
+* `calories` : int
+* `allergies` : ArrayList<String>
+* `mealCategory` : MealCategory
+* `url` : String
+* `dateAdded` : LocaleDate
```

Comment:
Appreciate the elaboration on the attributes but not sure if its necessary since the information can be found from the above class diagram.